test/integration/edition_edit_test.rb

```diff
@@ -108,54 +83,171 @@ class EditionEditTest < IntegrationTest
-        assert page.has_text?("For example: https://www.gov.uk/redirect-to-replacement-page")
-        assert page.has_css?(".govuk-input", count: 1)
+      context "when state is publish" do
+        setup do
+          visit_edition_in_published
+          click_link("Unpublish")
+        end
+
+        should "show 'Unpublish' header and 'Continue' button" do
+          within :css, ".gem-c-heading" do
+            assert page.has_text?("Unpublish")
+          end
+          assert page.has_button?("Continue")
+        end
+
+        should "show 'cannot be undone' banner" do
+          assert page.has_text?("If you unpublish a page from GOV.UK it cannot be undone.")
+        end
+
+        should "show 'Redirect to URL' text, input box and example text" do
+          assert page.has_text?("Redirect to URL")
+          assert page.has_text?("For example: https://www.gov.uk/redirect-to-replacement-page")
+          assert page.has_css?(".govuk-input", count: 1)
+        end
+
+        should "navigate to 'confirm-unpublish' page when 'Continue' button is clicked" do
+          click_button("Continue")
+          assert_equal(page.current_path, "/editions/#{@published_edition.id}/unpublish/confirm-unpublish")
+        end
       end
 
-      should "navigate to 'confirm-unpublish' page when clicked on 'Continue' button" do
-        click_button("Continue")
-        assert_equal(page.current_path, "/editions/#{@edition.id}/unpublish/confirm-unpublish")
+      context "when state is not publish" do
+        setup do
+          edition = FactoryBot.create(:edition, state: "draft")
+          visit edition_path(edition)
+        end
+
+        should "not show unpublish tab" do
+          assert page.has_no_text?("Unpublish")
+        end
       end
     end
+  end
 
-    context "confirm unpublish" do
+  context "admin tab" do
+    context "do not have required permissions" do
       setup do
-        click_link("Unpublish")
-        click_button("Continue")
+        login_as(FactoryBot.create(:user, name: "Stub User"))
+        visit_edition_in_draft
       end
 
-      should "show 'Unpublish' header and document title" do
-        assert page.has_text?("Unpublish")
-        assert page.has_text?(@edition.title.to_s)
+      should "not show when user is not govuk editor or welsh editor" do
+        assert page.has_no_text?("Admin")
       end
 
-      should "show 'cannot be undone' banner" do
-        assert page.has_text?("If you unpublish a page from GOV.UK it cannot be undone.")
+      should "not show when user is welsh editor and edition is not welsh" do
+        login_as(FactoryBot.create(:user, :welsh_editor, name: "Stub User"))
+        visit_edition_in_draft
+
+        assert page.has_no_text?("Admin")
       end
+    end
+
+    context "has required permissions" do
+      setup do
+        login_as(FactoryBot.create(:user, :govuk_editor, name: "Stub User"))
+      end
+
+      context "when state is not 'fact_check'" do
```

Comment:
We have quotes around "fact_check" but not around "draft" and "published"—we should consistently use or not use quotes.